pkg/rules/handlers.go

```diff
@@ -0,0 +1,268 @@
+package rules
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/canonical/identity-platform-admin-ui/internal/http/types"
+	"github.com/canonical/identity-platform-admin-ui/internal/logging"
+
+	"github.com/go-chi/chi/v5"
+	oathkeeper "github.com/ory/oathkeeper-client-go"
+)
+
+type API struct {
+	service ServiceInterface
+
+	logger logging.LoggerInterface
+}
+
+func (a *API) RegisterEndpoints(mux *chi.Mux) {
+	mux.Get("/api/v0/rules", a.handleList)
+	mux.Get("/api/v0/rules/{id}", a.handleDetail)
+	mux.Post("/api/v0/rules", a.handleCreate)
+	mux.Put("/api/v0/rules", a.handleUpdate)
```

Comment:
unfortunately this needs the id in the path

Reply:
Done. In case of a mismatch between the id url parameter, and the id rule field the request returns with an error.